### PR TITLE
`FastHGT`: Minor additions

### PR DESCRIPTION
**Pull request by puririshi98**
should be ready to merge

### CODE REVIEW

File: torch_geometric/utils/sort_edge_index.py [hunk 1]
```diff
@@ -90,3 +96,61 @@ def sort_edge_index(
         return edge_index, [e[perm] for e in edge_attr]
 
     return edge_index
+
+
+def construct_edge_index(
```
Comment (rusty1s) [inline torch_geometric/utils/sort_edge_index.py:101]:
```suggestion
def construct_bipartite_edge_index(
```

Comment (rusty1s) [inline torch_geometric/utils/sort_edge_index.py:101]: Can we move this to `utils/hetero.py`?

Comment (rusty1s) [inline torch_geometric/utils/sort_edge_index.py:101]: Can we add a test for this?

File: torch_geometric/nn/conv/fast_hgt_conv.py [hunk 2]
```diff
@@ -213,8 +213,9 @@ def forward(
         q, dst_offset = self._cat(q_dict)
         k, v, src_offset = self._construct_src_node_feat(k_dict, v_dict)
 
-        edge_index, edge_attr = self._construct_edge_index(
-            edge_index_dict, src_offset, dst_offset)
+        edge_index, edge_attr = construct_edge_index(edge_index_dict,
```
Comment (rusty1s) [inline torch_geometric/nn/conv/fast_hgt_conv.py:216]: Drop the method in the class.

File: test/nn/conv/test_hgt_conv.py [hunk 3]
```diff
@@ -174,6 +175,32 @@ def test_hgt_conv_out_of_place():
...
+    data = HeteroData()
+    data['v0'].x = torch.randn(5, 4)
+    data['v1'].x = torch.randn(5, 4)
+    data[('v0', 'e1', 'v0')].edge_index = torch.randint(high=5, size=(2, 10))
+    fast_net = FastHGTConv(4, 2, data.metadata())
+    og_net = HGTConv(4, 2, data.metadata())
+    x_dict = data.collect('x')
+    # make params match
+    for og_param, my_param in zip(og_net.parameters(), fast_net.parameters()):
+        try:
```
Comment (rusty1s) [inline test/nn/conv/test_hgt_conv.py:189]: What do we catch here?

File: test/nn/conv/test_hgt_conv.py [hunk 4]
```diff
@@ -174,6 +175,32 @@ def test_hgt_conv_out_of_place():
...
+            pass
+        try:
+            og_param.data = torch.ones_like(og_param.data)
+        except:  # noqa
+            pass
+
+    edge_index_dict = data.collect('edge_index')
+    our_o = list(fast_net(x_dict, edge_index_dict).values())[0]
+    og_o = list(og_net(x_dict, edge_index_dict).values())[0]
+    assert torch.allclose(our_o, og_o, atol=3e-3)
```
Comment (rusty1s) [inline test/nn/conv/test_hgt_conv.py:201]: How does `out_dict['v1']` look like?

**Comment by berkekisin** [comment]
hello we were also working on the fast HGT implementation. think i have some useful additions for this:
1) https://github.com/pyg-team/pytorch_geometric/blob/023d5938a0344ca8cabcaa0e855e32df455c6a38/torch_geometric/nn/conv/fast_hgt_conv.py#L130
think there is a bug here, it should be self.v_rel instead of self.k_rel.
2) https://github.com/pyg-team/pytorch_geometric/blob/0e439f38b674e551491c99e5a6ddbc7a62f4cbf4/torch_geometric/nn/conv/fast_hgt_conv.py#L51-L53
The HeteroLinears for k,q and v can be merged into one big HeteroLinear like this
`self.kqv_lin = HeteroDictLinear(self.in_channels, self.out_channels*3)`. then we can split them.

**Comment by berkekisin** [comment]
hello we were also working on the fast HGT implementation. think i have some useful additions for this:
1) https://github.com/pyg-team/pytorch_geometric/blob/023d5938a0344ca8cabcaa0e855e32df455c6a38/torch_geometric/nn/conv/fast_hgt_conv.py#L130
think there is a bug here, it should be self.v_rel instead of self.k_rel.
2) https://github.com/pyg-team/pytorch_geometric/blob/0e439f38b674e551491c99e5a6ddbc7a62f4cbf4/torch_geometric/nn/conv/fast_hgt_conv.py#L51-L53
The HeteroLinears for k,q and v can be merged into one big HeteroLinear like this
`self.kqv_lin = HeteroDictLinear(self.in_channels, self.out_channels*3)`. then we can split them.

**Comment by berkekisin** [comment]
hello we were also working on the fast HGT implementation. think i have some useful additions for this:
1) https://github.com/pyg-team/pytorch_geometric/blob/023d5938a0344ca8cabcaa0e855e32df455c6a38/torch_geometric/nn/conv/fast_hgt_conv.py#L130
think there is a bug here, it should be self.v_rel instead of self.k_rel.
2) https://github.com/pyg-team/pytorch_geometric/blob/0e439f38b674e551491c99e5a6ddbc7a62f4cbf4/torch_geometric/nn/conv/fast_hgt_conv.py#L51-L53
The HeteroLinears for k,q and v can be merged into one big HeteroLinear like this
`self.kqv_lin = HeteroDictLinear(self.in_channels, self.out_channels*3)`. then we can split them.

**Comment by puririshi98** [comment]
@berkekisin your comments are very useful thanks! I made your suggested change and have added you as a reviewer in case you have any additional comments which are always appreciated :).

**Pull request merged**